instrumentation/hibernate/hibernate-4.0/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/hibernate/v4_0/EntityManagerTest.java

```diff
@@ -0,0 +1,370 @@
+                            satisfies(
+                                SemanticAttributes.DB_STATEMENT,
+                                val -> val.isInstanceOf(String.class)),
+                            satisfies(
+                                SemanticAttributes.DB_OPERATION,
+                                val -> val.isInstanceOf(String.class)),
+                            equalTo(SemanticAttributes.DB_SQL_TABLE, "Value")));
+
+          } else {
+            trace.hasSpansSatisfyingExactly(
+                span ->
+                    span.hasName("parent")
+                        .hasKind(INTERNAL)
+                        .hasNoParent()
+                        .hasAttributes(Attributes.empty()),
+                span ->
+                    span.hasName("Session." + action + " " + parameter.resource)
+                        .hasKind(INTERNAL)
+                        .hasParent(trace.getSpan(0))
+                        .hasAttributesSatisfyingExactly(
+                            satisfies(
+                                stringKey("hibernate.session_id"),
+                                val -> val.isInstanceOf(String.class))),
+                span ->
+                    span.hasKind(CLIENT)
+                        .hasParent(trace.getSpan(1))
+                        .hasAttributesSatisfyingExactly(
+                            equalTo(SemanticAttributes.DB_SYSTEM, "h2"),
+                            equalTo(SemanticAttributes.DB_NAME, "db1"),
+                            equalTo(SemanticAttributes.DB_USER, "sa"),
+                            equalTo(SemanticAttributes.DB_CONNECTION_STRING, "h2:mem:"),
+                            satisfies(
+                                SemanticAttributes.DB_STATEMENT,
+                                val -> val.isInstanceOf(String.class)),
+                            satisfies(
+                                SemanticAttributes.DB_OPERATION,
+                                val -> val.isInstanceOf(String.class)),
+                            equalTo(SemanticAttributes.DB_SQL_TABLE, "Value")),
+                span ->
+                    span.hasName("Transaction.commit")
+                        .hasKind(INTERNAL)
+                        .hasParent(trace.getSpan(0))
+                        .hasAttributesSatisfyingExactly(
+                            equalTo(
+                                stringKey("hibernate.session_id"),
+                                trace
+                                    .getSpan(1)
+                                    .getAttributes()
+                                    .get(stringKey("hibernate.session_id")))));
+          }
+        });
+  }
+
+  private static Stream<Arguments> provideArgumentsHibernateActionParameters() {
+    return Stream.of(
+        Arguments.of(
+            named(
+                "lock",
+                new Parameter(
+                    "lock",
+                    Value.class.getName(),
```

Comment:
The `parameter.resource` is always, `Value.class.getName()`, WDYT of removing that field?